public_html/wp-content/plugins/multi-event-sponsors/classes/mes-sponsor.php

```diff
@@ -63,13 +64,129 @@ public function create_post_type() {
 				'with_front' => false,
 			),
 			'query_var'       => true,
+			'menu_icon'       => 'dashicons-heart',
 			'supports'        => array( 'title', 'editor', 'author', 'excerpt', 'revisions', 'thumbnail' ),
 			'taxonomies'      => array( MES_Region::TAXONOMY_SLUG ),
 		);
 
 		register_post_type( self::POST_TYPE_SLUG, $post_type_params );
 	}
 
+	/**
+	 * Register endpoints for the REST API.
+	 */
+	public function register_routes() : void {
+		register_rest_route(
+			'multi-event-sponsors/v1',
+			'/push-to-active-camps',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'push_to_active_camps' ),
+					'permission_callback' => function() {
+						return current_user_can( 'manage_network' );
+					},
+				),
+			)
+		);
+	}
+
+	/**
+	 * Copy the contents of MES posts to their forked posts on active camp sites.
+	 */
+	public function push_to_active_camps( WP_REST_Request $request ) : array {
+		$edited_posts = array();
+		$source_post  = get_post( $request->get_param( 'sponsorId' ) );
+		$wordcamps    = get_wordcamps( array(
+			'post_status' => array_merge(
+				// Any status where the camp might have a site created, and the event isn't over.
+				WordCamp_Loader::get_pre_planning_post_statuses(),
+				WordCamp_Loader::get_active_wordcamp_statuses(),
+			),
+		) );
+
+		foreach ( $wordcamps as $wordcamp ) {
+			if ( ! $wordcamp->meta['_site_id'][0] ) {
+				continue;
+			}
+
+			switch_to_blog( $wordcamp->meta['_site_id'][0] );
+
+			$fork_post = get_posts( array(
+				'post_type'      => 'wcb_sponsor',
+				'meta_key'       => '_mes_id',
+				'meta_value'     => $source_post->ID,
+				'meta_compare'   => '=',
+			) );
+			$fork_post = array_pop( $fork_post );
+
+			if ( ! $fork_post ) {
+				restore_current_blog();
+				continue;
+			}
+
+			// Organizers might have made changes that are specific to this camp, so don't overwrite them.
+			if ( self::post_has_been_edited( $fork_post->ID ) ) {
+				$edited_posts[] = array(
+					// Can't use `get_post_edit_link()` because the `wcb_sponsor` post type isn't registered on Central or while switching blogs.
+					'edit_url'  => admin_url( "post.php?post={$fork_post->ID}&action=edit" ),
+					'site_name' => get_wordcamp_name(),
+				);
```

Comment:
No I meant `unchanged_posts`, because these posts have not been changed by this code. `changed_posts` would have the same issue as `edited_posts` does. I don't think that requires a logic refactor, unless I'm really misunderstanding the code here.

Actually, `forked_posts` has a similar problem, because to me forked implies changed too, but the `forked_posts` are not necessarily changed… but that's not as confusing in context.

Reply:
Ah, you meant changed by the _code_, while I meant changed by the _user_. So maybe `$forks_edited_by_user[]`, to clarify both: a) which post type; and b) who edited it? 

The forking I'm referring to is when the `wcb_sponsor` posts are created on new sites, based on the canonical `mes` posts, similar to forking a Git repo. To me forking doesn't imply changing, just the _option_ to change. Maybe it's best to not use a metaphor anyway, though, and instead do something like `$wcb_sponsor_posts[]`. That feels vaguely inelegant to me, but maybe less so? 🤷🏻‍♂️

Reply:
Yeah, the `fork` is vague enough that while it took a second to understand your meaning, it's fine since it's consistent (I understood the git reference, maybe I should have said "to me fork implies it's likely to have been changed"). I don't think you need to worry about changing that word.

What about `skipped_posts`? If you don't like that, I'm okay with the longer `forks_edited_by_user`.

Reply:
`skipped_posts` sounds good to me 👍🏻